include/dlaf/matrix/matrix_const.tpp

```diff
@@ -72,5 +72,29 @@ void Matrix<const T, D>::setUpTiles(const memory::MemoryView<ElementType, D>& me
   }
 }
 
+template <class T, Device D>
+Matrix<const T, D>::Matrix(Matrix<const T, D>& mat, const SubPipelineTag)
+    : MatrixBase(mat.distribution()) {
+  setUpSubPipelines(mat);
+}
+
+template <class T, Device D>
+void Matrix<const T, D>::setUpSubPipelines(Matrix<const T, D>& mat) noexcept {
+  namespace ex = pika::execution::experimental;
+
+  // TODO: Optimize read-after-read. This is currently forced to access the base
```

Comment:
Is this something you will solve or postpone to a later PR?

Reply:
Definitely postponing. Opened https://github.com/eth-cscs/DLA-Future/issues/923 to track it.